src/utils/image_server.cljs

```diff
@@ -58,3 +60,55 @@
        "&clock="
        clock
        "&addRing=1"))
+
+(defn get-account-qr-image-uri
+  [{:keys [key-uid public-key port qr-size]}]
+  (let [profile-qr-url         (str status-profile-base-url public-key)
+        base-64-qr-url         (js/btoa profile-qr-url)
+        profile-image-type     "large"
+        error-correction-level 4
+        superimpose-profile?   true
+        media-server-url       (str image-server-uri-prefix
+                                    port
+                                    generate-qr-action
+                                    "?level="
+                                    error-correction-level
+                                    "&url="
+                                    base-64-qr-url
+                                    "&keyUid="
+                                    key-uid
+                                    "&allowProfileImage="
+                                    superimpose-profile?
+                                    "&size="
+                                    qr-size
+                                    "&imageName="
+                                    profile-image-type)]
+    media-server-url))
+
+;for an input string label of error correction level
+;this fn must return the appropriate corresponding index #
+(defn get-error-correction-index-for-level
+  [level]
+  (case level
```

Comment:
Here you can use `(case (keyword level) ...`, so that this function works with string and keyword inputs. This is a common approach to avoid the common bug that the function is passed a string `"highest"` and no match is found, even though it should.

In terms of naming, it's more idiomatic to name this function `correction-level->index` and then the comment above the function becomes superfluous (at least to me).

Reply:
addressed here : https://github.com/status-im/status-mobile/pull/15755/commits/ddbe0e089796e1dead6cddb88adda90dc4acd8ba